pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1537,6 +1537,161 @@ type ObjectZoneSpec struct {
 	DataPool PoolSpec `json:"dataPool"`
 }
 
+// CephBucketTopic represents a Ceph Object Topic for Bucket Notifications
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+type CephBucketTopic struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              BucketTopicSpec `json:"spec"`
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +optional
+	Status *BucketTopicStatus `json:"status,omitempty"`
+}
+
+// BucketTopicStatus represents the Status of a CephBucketTopic
+type BucketTopicStatus struct {
+	// +optional
+	Phase string `json:"phase,omitempty"`
+	// The ARN of the topic generated by the RGW
+	// +optional
+	// +nullable
+	ARN *string `json:"ARN,omitempty"`
+}
+
+// CephBucketTopicList represents a list Ceph Object Store Bucket Notification Topics
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type CephBucketTopicList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []CephBucketTopic `json:"items"`
+}
+
+// BucketTopicSpec represent the spec of a Bucket Topic
+type BucketTopicSpec struct {
+	// The URI of an endpoint to push notification to
+	Endpoint string `json:"endpoint"`
+	// The name of the object store on which to define the topic
+	ObjectStoreName string `json:"objectStoreName"`
+	// Data which is sent in each event
+	// +optional
+	OpaqueData string `json:"opaqueData,omitempty"`
+	// Indication whether notifications to this endpoint are persistent or not
+	// +optional
+	Persistent bool `json:"persistent,omitempty"`
+	// Spec of HTTP endpoint
+	// +optional
+	HTTP *HTTPEndpointSpec `json:"http"`
+	// Spec of AMQP endpoint
+	// +optional
+	AMQP *AMQPEndpointSpec `json:"amqp"`
+	// Spec of Kafka endpoint
+	// +optional
+	Kafka *KafkaEndpointSpec `json:"kafka"`
+}
+
+// HTTPEndpointSpec represent the spec of an HTTP endpoint of a Bucket Topic
+type HTTPEndpointSpec struct {
+	// Indicate whether the server certificate is validated by the client or not
+	// +optional
+	VerifySSL bool `json:"verifySSL,omitempty"`
```

Comment:
Let's use TLS instead of SSL.

Reply:
this is the term used for this parameter in the RGW docs

Reply:
I think if we are directing users to outside documentation (i.e., Ceph), it is best for users to have the same terminology between the two.